lib/shoryuken/manager.rb

```diff
@@ -78,6 +78,7 @@ def processor_done(queue, processor)
     def processor_died(processor, reason)
       watchdog("Manager#processor_died died") do
         logger.error { "Process died, reason: #{reason}" unless reason.to_s.empty? }
+        reason.backtrace.each { |bt| logger.error { bt } } unless reason.backtrace.nil?
```

Comment:
I've not encountered `reason = nil` but I haven't checked/tested extensively, `backtrace = nil` was certainly true for some failures.  I would be happy to refactor as you suggested.

It's been so long that I honestly can't remember the specific problem that drove me to suggest this change, but I'll try recreating and provide a sample output when I get some time.  However, I would argue regardless of how far back it goes, something is better than nothing.